lib/demo_elixir_phoenix_web/controllers/page_controller.ex

```diff
@@ -17,73 +17,136 @@ defmodule DemoElixirPhoenixWeb.PageController do
       )
 
     conn = KindeClientSDK.login(conn, client)
-
     response = KindeClientSDK.get_all_data(conn)
 
-    render(conn, "index.html", response: response.token)
+    render(conn, "index.html", response: response)
   end
 
   def callback(conn, _params) do
-    KindeClientSDK.get_kinde_client(conn)
+    {conn, _client} = KindeClientSDK.get_token(conn)
+    user = KindeClientSDK.get_user_detail(conn)
 
-    {conn, client} = KindeClientSDK.get_token(conn)
-
-    render(conn, "index.html", response: nil)
+    if KindeClientSDK.authenticated?(conn) do
+      render(conn, "callback.html", user: user)
+    else
+      render(conn, "index.html", user: user)
+    end
   end
 
   def get_claims(conn, _) do
-    response = KindeClientSDK.get_claims(conn)
-    render(conn, "index.html", response: response)
+    if KindeClientSDK.authenticated?(conn) do
+      KindeClientSDK.get_kinde_client(conn)
+      {conn, _client} = KindeClientSDK.get_token(conn)
+      response = KindeClientSDK.get_claim(conn, "iss")
+      render(conn, "callback.html", response: response)
+    else
+      render(conn, "index.html", response: nil)
+    end
   end
 
   def get_claim(conn, _) do
-    ## you can change the "iss" to any other claim-object-key
-    response = KindeClientSDK.get_claim(conn, "iss")
-    render(conn, "index.html", response: response)
+    if KindeClientSDK.authenticated?(conn) do
+      KindeClientSDK.get_kinde_client(conn)
+      {conn, _client} = KindeClientSDK.get_token(conn)
+      response = KindeClientSDK.get_claim(conn, "iss")
+      render(conn, "callback.html", response: response)
+    else
+      render(conn, "index.html", response: nil)
+    end
   end
 
   # Tip: For this action to work, remember to do the PKCE login first
   def get_claim_from_id_token(conn, _) do
-    ## you can change the "iss" to any other claim-object-key
-    response = KindeClientSDK.get_claim(conn, "iss", :id_token)
-    render(conn, "index.html", response: response)
+    response = nil
```

Comment:
Just small code refactor: we can even remove this variable and just pass nil in else part like `response: nil`